api/v1alpha1/vaultsecret_types.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"net/url"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// VaultSecretSpec defines the desired state of VaultSecret
+type VaultSecretSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Url of the Vault instance.
+	// +kubebuilder:validation:Required
+	Url string `json:"url,omitempty"`
+	// RefreshPeriod if specified, the operator will refresh the secret with the given frequency.
+	// Defaults to five minutes, and must be at least one minute.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="5m"
+	RefreshPeriod *metav1.Duration `json:"refreshPeriod,omitempty"`
```

Comment:
I'd like to discuss this feature:
1. most vault secrets tells you when they are going to expire, so the operator should made the choice based on that info.
2.  if we want to support an interval-based refresh, then there should be the option of never update. And perhaps that should be the default.
3. shouldn't this be at the secret level?

Reply:
The intention is to control the pull policy. It seems irrelevant to worry about the Vault configuration's expiration since users can update the secret at any time. I think keeping this pull interval completely separate is simple and makes the most sense to the user.

Reply:
you are thinking mostly about the kv secret store, but that is really a minor use case (although it's used in all of the demos because it's easy to set up), but all of the other secret engines have expiration. I'd like to discuss this more.